### ISSUE TITLE
watchForReceipt throws an error if the client is not connected

### ISSUE THREAD

**Issue by akaNightmare** [stomp-js/rx-stomp#484]
Hello.

```
async function test(receipt) {
  rxClient.publish({
    // other params
    headers: { receipt },
  });
  return new Promise(resolve => rxClient.watchForReceipt(receipt, resolve));
}
```
throws an error `TypeError: Cannot read property 'watchForReceipt' of undefined` of the client is not connected

Thanks.

**Comment by kum-deepak** [comment]
To be fixed in the next release.

**Comment by akaNightmare** [comment]
@kum-deepak any updates?

**Comment by kum-deepak** [comment]
I realized there is another way to watch for receipts, which works better with RxStomp and does not suffer from the issue you reported.

Two changes:

- It uses [unhandledReceipts$](https://stomp-js.github.io/api-docs/latest/classes/RxStomp.html#unhandledReceipts$).
- The promise for receipt is set up before the actual call.

```typescript
  const promiseReceipt = (receiptId: string) =>
    firstValueFrom(
      rxStomp.unhandledReceipts$.pipe(
        filter(frame => frame.headers['receipt-id'] === receiptId)
      )
    );
// ....
    // Now publish to the same queue with the receipt request
    const publishReceipt = 'publish-receipt';
    const promisePublishReceipt = promiseReceipt(publishReceipt);
    rxStomp.publish({
      destination: queueName,
      body: msg,
      headers: { receipt: publishReceipt },
    });
    await promisePublishReceipt;


```

Try this out. Based on that, I will update the documentation.

**Comment by akaNightmare** [comment]
thanks. Let's keep it open until the issue will be fixed

**Comment by kum-deepak** [comment]
Did it work for you?

**Comment by akaNightmare** [comment]
this is working as expected, thank you.

**Comment by kum-deepak** [comment]
I just now released  @stomp/rx-stomp@2.0.0-beta1. The newly added function is [asyncReceipt](https://stomp-js.github.io/api-docs/develop/classes/RxStomp.html#asyncReceipt)

**Comment by kum-deepak** [comment]
Released v2.0.0
